java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,257 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Arguments for {@link glide.api.commands.SortedSetBaseCommands#zrange} and {@link
+ * glide.api.commands.SortedSetBaseCommands#zrangeWithScores}
+ *
+ * @see <a href="https://redis.io/commands/zrange/">redis.io</a>
+ */
+public class RangeOptions {
+
+    public interface ScoreRange {
+        String toArgs();
+    }
+
+    /** Enumeration representing numeric positive and negative infinity bounds for a sorted set. */
+    @RequiredArgsConstructor
+    public enum InfScoreBound implements ScoreRange {
+        POSITIVE_INFINITY("+inf"),
+        NEGATIVE_INFINITY("-inf");
+
+        private final String redisApi;
+
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific numeric score boundary in a sorted set. */
+    public static class ScoreBoundary implements ScoreRange {
+        /** The score value. */
+        private final double bound;
+
+        /** Whether the score value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public ScoreBoundary(double bound, boolean isInclusive) {
+            this.bound = bound;
+            this.isInclusive = isInclusive;
+        }
+
+        public ScoreBoundary(double bound) {
+            this.bound = bound;
+            this.isInclusive = true;
+        }
+
+        /** Convert the score boundary to the Redis protocol format. */
+        public String toArgs() {
+            return this.isInclusive ? String.valueOf(this.bound) : "(" + this.bound;
+        }
+    }
+
+    public interface LexRange {
+        String toArgs();
+    }
+
+    /**
+     * Enumeration representing lexicographic positive and negative infinity bounds for sorted set.
+     */
+    @RequiredArgsConstructor
+    public enum InfLexBound implements LexRange {
+        POSITIVE_INFINITY("+"),
+        NEGATIVE_INFINITY("-");
+
+        private final String redisApi;
+
+        @Override
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific lexicographic boundary in a sorted set. */
+    public static class LexBoundary implements LexRange {
+        /** The lex value. */
+        private final String value;
+
+        /** Whether the lex value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public LexBoundary(@NonNull String value, boolean isInclusive) {
```

Comment:
docs for constructors

Reply:
I don't think there is a need for that the instance variables have documentation plus the class declaration has documentation.